akka-stream/src/main/scala/akka/stream/impl/ActorMaterializerImpl.scala

```diff
@@ -255,7 +254,7 @@ private[akka] class SubFusingActorMaterializerImpl(
 
     // If the timeout hits it is likely because one of the streams stopped between looking at the list
     // of children and asking it for a snapshot. We retry the entire snapshot in that case
-    retry(() => takeSnapshot(), 3, Duration.Zero)
+    retry(() => takeSnapshot(), 3)
```

Comment:
The new implementation will create a closure, so change to this one which doesn't.